lib/Backend/FlowGraph.cpp

```diff
@@ -4541,38 +4549,96 @@ BasicBlock::CheckLegalityAndFoldPathDepBranches(GlobOpt* globOpt)
     // If we find a Ld_A, we find if the src1 already has a value in the local valueTable, if not we use the value from the pred's valueTable merge
     // For other instrs, we assign null value
     GlobHashTable * localSymToValueMap = GlobHashTable::New(globOpt->alloc, 8);
+    BVSparse<JitArenaAllocator> currentPathDefines(globOpt->alloc);
+    IR::Instr *currentInlineeEnd = nullptr, *unskippedInlineeEnd = nullptr;
+
+    auto UpdateValueForCopyTypeInstr = [&](IR::Instr *instr) {
+        if (instr->GetSrc1()->GetStackSym())
+        {
+            StackSym* src1Sym = instr->GetSrc1()->GetStackSym();
+            // MGTODO : this is needed for loads of undefined/true/false in prepass
+            // We dont aggressively transfer their values yet, once it is enabled this check can be removed
```

Comment:
>We dont aggressively transfer their values yet [](start = 15, length = 46)

As you had pointed out, we can transfer values aggressively in prepass along the path we are on, no?

Even if we do need to look at the instrdef, we should ideally follow the singledef chain past all the intermediate Ld_A's, and not just stop at the first instrDef which is Ld_A.

Reply:
Added to commit to look up this chain. We have to lookup this chain because the local valuetable in prepass or not, might not have seen constants defined in the top of the function of top of the loop. We dont preserve values in localValueTable across optimization trials.